src/cli_common.c

```diff
@@ -356,6 +356,7 @@ void parseRedisUri(const char *uri, const char* tool_name, cliConnInfo *connInfo
             connInfo->hostport = atoi(port + 1);
             host = port - 1;
         }
+        sdsfree(connInfo->hostip);
```

Comment:
Small leak, address sanitizer complains

Reply:
@oranagra I added sanitizers to PR CI (With same configs with regular PR CI : ./runtest --verbose --tags -slow).

I also realized we don't check exit code of Redis when we kill it. So, we were missing some sanitizer warnings that is printed on exit. I had to fix them and added checks to search log on exit for sanitizer errors, similar to valgrind build. Please take a look at last couple of commits.